runtime/doc/autocmd.txt

```diff
@@ -729,7 +729,22 @@ MenuPopup			Just before showing the popup menu (under the
 					o	Operator-pending
 					i	Insert
 					c	Command line
-							*OptionSet*
+							*ModeChanged*
+ModeChanged			After changing the mode. The pattern is
+				matched against `'old_mode:new_mode'`, for
+				example match against `i:*` to simulate
+				|InsertLeave|.
+				The following values of |v:event| are set:
+					old_mode The mode before it changed.
+					new_mode The new mode as also returned
+					         by |mode()|.
```

Comment:
In case you found out you original Vim patch was wrong, this should `mode(1)` instead of `mode()`.